features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/utils/MiscUtils.kt

```diff
@@ -12,14 +12,62 @@ import android.graphics.Point
 import android.os.Build
 import android.util.TypedValue
 import android.view.WindowManager
+import com.datadog.android.api.InternalLogger
 import com.datadog.android.sessionreplay.internal.recorder.GlobalBounds
 import com.datadog.android.sessionreplay.internal.recorder.SystemInformation
 import com.datadog.android.sessionreplay.internal.recorder.densityNormalized
 import com.datadog.android.sessionreplay.utils.StringUtils
+import com.google.gson.JsonObject
+import com.google.gson.JsonParser
+import com.google.gson.JsonSyntaxException
 
 internal object MiscUtils {
 
     internal const val OPAQUE_ALPHA_VALUE = 255
+    private const val GET_STRING_FROM_JSON_ERROR = "Error getting string property from json"
+    internal const val DESERIALIZE_JSON_ERROR = "Error deserializing json object"
+
+    internal fun safeDeserializeToJsonObject(internalLogger: InternalLogger, jsonByteArray: ByteArray): JsonObject? {
+        if (jsonByteArray.isEmpty()) return null
+
+        val jsonString = String(jsonByteArray)
+        return try {
+            @Suppress("UnsafeThirdPartyFunctionCall")
```

Comment:
> [!WARNING]
> Why are you suppressing the check here? You should either catch everything possible, since it's a `safe…` function, or add a comment to tell why we're suppressing it